src/backend/commands/createas.c

```diff
@@ -1335,7 +1373,12 @@ check_ivm_restriction_walker(Node *node, check_ivm_restriction_context *ctx, int
 							(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 							 errmsg("this query is not allowed on incrementally maintainable materialized view"),
 							 errhint("subquery with outer join is not supported")));
+
+				ctx->has_exists_subquery = true;
+				if (IsA(subselect->jointree, FromExpr))
+					ctx->join_quals = lappend(ctx->join_quals, subselect->jointree->quals);
```

Comment:
Actually, subselect->jointree->quals represents WHERE clause rather than JOIN quals, it seems better for me to add a new field like ctx->where_quals_in_exists.

In addition, anyway, this value is not used in your code. Is this hunk necessary?

Reply:
Thank you for pointing out my mistake. This processing is not needed. i will remove it.